samples/dotnet/FileCompression/FileCompressionSkill.cs

```diff
@@ -0,0 +1,158 @@
+// string sourceFilePath = "FileToCompress.txt";
+// string destinationFilePath = "CompressedFile.zip";
+// var variables = new ContextVariables(sourceFilePath);
+// variables.Set(FileCompressionSkill.Parameters.DestinationFilePath, destinationFilePath);
+// await kernel.RunAsync(variables, fileCompression["CompressFileAsync"]);
+//**********************************************************************************************************************
+
+/// <summary>
+/// Skill for compressing and decompressing files.
+/// </summary>
+public class FileCompressionSkill
+{
+    /// <summary>
+    /// Parameter names.
+    /// <see cref="ContextVariables"/>
+    /// </summary>
+    public static class Parameters
+    {
+        /// <summary>
+        /// Directory to which to extract compressed file's data.
+        /// </summary>
+        public const string DestinationDirectoryPath = "destinationDirectoryPath";
+
+        /// <summary>
+        /// File path where to save compressed data.
+        /// </summary>
+        public const string DestinationFilePath = "destinationFilePath";
+    }
+
+    private readonly IFileCompressor _fileCompressor;
+    private readonly ILogger<FileCompressionSkill> _logger;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="FileCompressionSkill"/> class.
+    /// </summary>
+    /// <param name="fileCompressor">File compressor implementation</param>
+    /// <param name="logger">Optional logger</param>
+    public FileCompressionSkill(IFileCompressor fileCompressor, ILogger<FileCompressionSkill>? logger = null)
+    {
+        this._fileCompressor = fileCompressor ?? throw new ArgumentNullException(nameof(fileCompressor));
+        this._logger = logger ?? new NullLogger<FileCompressionSkill>();
+    }
+
+    /// <summary>
+    /// Compresses an input file to an output file.
+    /// </summary>
+    /// <param name="sourceFilePath">Path of file to compress</param>
+    /// <param name="context">Semantic Kernel context</param>
+    /// <returns>Path of created compressed file</returns>
+    /// <exception cref="KeyNotFoundException"></exception>
+    [SKFunction("Compresses an input file to an output file")]
+    [SKFunctionInput(Description = "Path of file to compress")]
+    [SKFunctionContextParameter(Name = Parameters.DestinationFilePath, Description = "Path of compressed file to create")]
+    public async Task<string> CompressFileAsync(string sourceFilePath, SKContext context)
+    {
+        this._logger.LogTrace($"{nameof(CompressFileAsync)} got called");
+
+        if (!context.Variables.Get(Parameters.DestinationFilePath, out string destinationFilePath))
+        {
+            const string errorMessage = $"Missing context variable {Parameters.DestinationFilePath} in {nameof(CompressFileAsync)}";
+            this._logger.LogError(errorMessage);
+            context.Fail(errorMessage);
+
+            throw new KeyNotFoundException(errorMessage);
```

Comment:
I believe we should only do a "context.Fail" when in the midst of an SK function, instead of throwing.

Reply:
Changing this.